autopilot/contractor.go

```diff
@@ -878,19 +887,30 @@ func (c *contractor) runRevisionBroadcast(ctx context.Context, w Worker, allCont
 	bh := cs.BlockHeight
 
 	successful, failed := 0, 0
-	for _, contract := range setContracts {
+	for _, contract := range allContracts {
 		// check whether broadcasting is necessary
 		timeSinceRevisionHeight := targetBlockTime * time.Duration(bh-contract.RevisionHeight)
 		timeSinceLastTry := time.Since(c.revisionLastBroadcast[contract.ID])
-		if contract.RevisionHeight == math.MaxUint64 || timeSinceRevisionHeight < c.revisionBroadcastInterval || timeSinceLastTry < c.revisionBroadcastInterval {
+		_, inSet := isInSet[contract.ID]
+		if !inSet || contract.RevisionHeight == math.MaxUint64 || timeSinceRevisionHeight < c.revisionBroadcastInterval || timeSinceLastTry < c.revisionBroadcastInterval/broadcastRevisionRetriesPerInterval {
 			continue // nothing to do
 		}
 
 		// remember that we tried to broadcast this contract now
 		c.revisionLastBroadcast[contract.ID] = time.Now()
 
+		// ignore contracts for which we weren't able to obtain a revision
+		if contract.Revision == nil {
```

Comment:
should we move this above `c.revisionLastBroadcast[contract.ID] = time.Now()`?

Reply:
No, this was intentional. If we failed to fetch a revision, the broadcast is considered tried and failed. We will retry periodically.